src/plugins/opensearch_dashboards_react/public/code_editor/code_editor.tsx

```diff
@@ -38,6 +38,14 @@ import { LIGHT_THEME, DARK_THEME } from './editor_theme';
 
 import './editor.scss';
 
+export interface LanguageSpecifiedConfiguration extends monaco.languages.LanguageConfiguration {
+  /**
+   * The language ID, meant to restrict the specified configuration for only this language. When
+   * not provided, will apply the language configuration for every language.
+   */
+  language?: string;
```

Comment:
what is this for? i think this should be a wrapper for the code editor but if there is logically differences then we add to osd-monaco

Reply:
this was to allow for only specific languages to be specified for configurations. i can remove it

Reply:
little bit confused, but you have more context than me. it seems like the only time this is set is in the loop where we check the language. and if we have to register the languages so we would still have to pass this prop. so as long as im registering the language then i can register the language configuration?

Reply:
so this interface is only used to specify which language the configuration should work for, its just an extra parameter that is checked within that loop to see if it should register the language configuration, or not